### PR TITLE
Refactor event handler

### PR DESCRIPTION
**Pull request by xellgrid**

### CODE REVIEW

File: qgrid/grid.py [hunk 1]
```diff
@@ -1554,7 +1528,7 @@ def add_row(self, row=None):
         if row is None:
             added_index = self._add_empty_row()
         else:
-            added_index = self._add_empty_row()
```
Comment (warm200) [inline qgrid/grid.py:1557]: seem add_row method is not referred in any way, the current [add_row ](https://github.com/xellgrid/XellGrid/pull/9/files#diff-57f868d4357688ef72d5fa006b0c83f0e6063beca5dd420316ebb3335344dbe3R1398)content type is actually triggering duplicate_last_row which is confusing. we might need to redesign this communication.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,9 @@
+six
+modin
```
Comment (warm200) [inline requirements.txt:2]: hmmm, i don't know if it's good idea to bring modin at this early stage, we still have work to do

**Review by warm200 — CHANGES_REQUESTED** [review]
please resolve the conflicts

File: qgrid/grid.py (hunk 1 above)
Comment (xellgrid) [reply qgrid/grid.py:1557]: _add_empty_row is an experimental function I added some time back to test communication between backend and front end.  I will try to find a better way handle "add rows" operations.

File: requirements.txt (hunk 2 above)
Comment (xellgrid) [reply requirements.txt:2]: agreed - we can remove it. It is just some experiment I did sometime back on Modin.

File: qgrid/grid.py [hunk 3]
```diff
@@ -1589,10 +1563,8 @@ def _duplicate_last_row(self):
 
     def _add_empty_row(self):
         print("add empty row")
-        print(self._selected_rows)
         df = self._df
         df1 = pd.DataFrame([[np.nan] * len(df.columns)], columns=df.columns)
-        print("step 2")
         df2 = df1.append(df, ignore_index=True)
```
[the remark this answers is not in the document]
Comment (xellgrid) [reply qgrid/grid.py:1568]: good catch - will use concat in the future

File: qgrid/grid.py [hunk 4]
```diff
@@ -967,6 +938,9 @@ def should_be_stringified(col_series):
                 if sort_column_name:
                     series_to_set = df[sort_column_name]
                 else:
+                    temp = self._get_col_series_from_df(
```
[the remark this answers is not in the document]
Comment (xellgrid) [reply qgrid/grid.py:941]: good catch - will remove it

**Comment by warm200** [comment]
@xellgrid  The conflicts is staright forward, so I just resolved it, see detail in  https://github.com/xellgrid/XellGrid/pull/9/commits/8f1b2df7b75ac4ef7bd565af15f1407906e12e0d

**Review by warm200 — APPROVED** [review]

**Pull request merged**